merlin/systems/dag/runtimes/triton.py

```diff
@@ -0,0 +1,279 @@
+                        node.parents_with_dependencies, input_col_name, "ensemble"
+                    )
+                    source_id = node_id_table.get(source, None)
+                    in_suffix = f"_{source_id}" if source_id is not None else ""
+
+                    if input_col_schema.is_list and input_col_schema.is_ragged:
+                        config_step.input_map[input_col_name + "__values"] = (
+                            input_col_name + "__values" + in_suffix
+                        )
+                        config_step.input_map[input_col_name + "__nnzs"] = (
+                            input_col_name + "__nnzs" + in_suffix
+                        )
+                    else:
+                        config_step.input_map[input_col_name] = input_col_name + in_suffix
+
+                for output_col_name, output_col_schema in node.output_schema.column_schemas.items():
+                    out_suffix = (
+                        f"_{node_id}" if node_id is not None and node_id < num_nodes - 1 else ""
+                    )
+
+                    if output_col_schema.is_list and output_col_schema.is_ragged:
+                        config_step.output_map[output_col_name + "__values"] = (
+                            output_col_name + "__values" + out_suffix
+                        )
+                        config_step.output_map[output_col_name + "__nnzs"] = (
+                            output_col_name + "__nnzs" + out_suffix
+                        )
+                    else:
+                        config_step.output_map[output_col_name] = output_col_name + out_suffix
+
+                ensemble_config.ensemble_scheduling.step.append(config_step)
+
+        # Write the ensemble config file
+        ensemble_path = os.path.join(path, name)
+        os.makedirs(ensemble_path, exist_ok=True)
+        os.makedirs(os.path.join(ensemble_path, str(version)), exist_ok=True)
+
+        config_path = os.path.join(ensemble_path, "config.pbtxt")
+        with open(config_path, "w", encoding="utf-8") as o:
+            text_format.PrintMessage(ensemble_config, o)
+
+        return (ensemble_config, node_configs)
+
+    def _find_column_source(self, upstream_nodes, column_name, backend):
+        source_node = None
+        for upstream_node in upstream_nodes:
+            if column_name in upstream_node.output_columns.names:
+                source_node = upstream_node
+                break
+
+        if source_node and not source_node.exportable(backend):
+            return self._find_column_source(
+                source_node.parents_with_dependencies, column_name, backend
+            )
+        else:
+            return source_node
+
+
+class TritonExecutorRuntime:
+    """Runtime for Triton.
+    This will run the DAG in a single Triton model and call out to other
+    Triton models for nodes that use any non-python backends.
+    """
+
+    def export(self, ensemble, path: str, version: int = 1, name: str = None):
```

Comment:
Should probably have a docstring for this method (pretty much identical to the `export` method above)